snowfakery/data_generator.py

```diff
@@ -176,6 +187,37 @@ def generate(
     return ExecutionSummary(parse_result, runtime_context)
 
 
+def process_plugins_options(
+    plugins: Mapping[str, SnowfakeryPlugin], raw_plugin_options: Mapping[str, object]
+) -> Mapping[str, object]:
+    """Replace option short names with fully qualified names
+       and convert types of options.
+    e.g. the option name that the user specifies on the CLI or API is just "orgname"
+         but we use the long name internally to aavoid clashing with the
+         user's variable names."""
+
+    allowed_options = collect_allowed_plugin_options(tuple(plugins.values()))
+    plugin_options = {}
+    for option in allowed_options:
+        option_long_name = option.name
+        option_short_name = option_long_name.rsplit(".", 1)[-1]
+        if option_short_name in raw_plugin_options:
+            plugin_options[option_long_name] = option.convert(
+                raw_plugin_options[option_short_name]
+            )
+        elif option_long_name in raw_plugin_options:
+            plugin_options[option_long_name] = option.convert(
+                raw_plugin_options[option_long_name]
+            )
+    return plugin_options
+
+
+def collect_allowed_plugin_options(plugins: Tuple) -> List[PluginOption]:
+    """Collect the list of every option allowed by every used plugin"""
+    all_option_lists = [getattr(plugin, "allowed_options", []) for plugin in plugins]
+    return functools.reduce(lambda x, y: x + y, all_option_lists, [])
```

Comment:
Perhaps slightly more readable:

sum(all_option_lists, start=[])

Reply:
Should there be an error if multiple plugins have an option with the same short name?